collector/perf_schema_replication_group_members.go

```diff
@@ -0,0 +1,160 @@
+			"channel_name, member_id, member_host, member_port, member_state, member_role, member_version. ",
+		[]string{"channel_name", "member_id", "member_host", "member_port", "member_state", "member_role", "member_version"}, nil,
+	)
+	performanceSchemaReplicationGroupMembersMemberDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, performanceSchema, "replication_group_member"),
+		"Information about the replication group member: "+
+			"channel_name, member_id, member_host, member_port, member_state. ",
+		[]string{"channel_name", "member_id", "member_host", "member_port", "member_state"}, nil,
+	)
+)
+
+var activeQuery string
+
+// ScrapeReplicationGroupMembers collects from `performance_schema.replication_group_members`.
+type ScrapePerfReplicationGroupMembers struct{}
+
+// Name of the Scraper. Should be unique.
+func (ScrapePerfReplicationGroupMembers) Name() string {
+	return performanceSchema + ".replication_group_members"
+}
+
+// Help describes the role of the Scraper.
+func (ScrapePerfReplicationGroupMembers) Help() string {
+	return "Collect metrics from performance_schema.replication_group_members"
+}
+
+// Version of MySQL from which scraper is available.
+func (ScrapePerfReplicationGroupMembers) Version() float64 {
+	return 5.7
+}
+
+// Scrape collects data from database connection and sends it over channel as prometheus metric.
+func (ScrapePerfReplicationGroupMembers) Scrape(ctx context.Context, db *sql.DB, ch chan<- prometheus.Metric, logger log.Logger) error {
+	perfReplicationGroupMembersRows, err := db.QueryContext(ctx, getPerfReplicationGroupMembersQuery(ctx, db, logger))
+	if err != nil {
+		return err
+	}
+	defer perfReplicationGroupMembersRows.Close()
+
+	var (
+		channelName   string
+		memberId      string
+		memberHost    string
+		memberPort    string
+		memberState   string
+		memberRole    string
+		memberVersion string
+	)
+
+	for perfReplicationGroupMembersRows.Next() {
+		if getPerfReplicationGroupMembersQuery(ctx, db, logger) == perfReplicationGroupMembersQueryWithAdditionalCols {
+			if err := perfReplicationGroupMembersRows.Scan(
+				&channelName, &memberId, &memberHost, &memberPort, &memberState, &memberRole, &memberVersion,
+			); err != nil {
+				return err
+			}
+			ch <- prometheus.MustNewConstMetric(
+				performanceSchemaReplicationGroupMembersMemberWithAdditionalColsDesc, prometheus.GaugeValue, 1,
+				channelName, memberId, memberHost, memberPort, memberState, memberRole, memberVersion,
+			)
+		} else {
+			if err := perfReplicationGroupMembersRows.Scan(
+				&channelName, &memberId, &memberHost, &memberPort, &memberState,
+			); err != nil {
+				return err
+			}
+			ch <- prometheus.MustNewConstMetric(
+				performanceSchemaReplicationGroupMembersMemberDesc, prometheus.GaugeValue, 1,
+				channelName, memberId, memberHost, memberPort, memberState,
+			)
+		}
+
+	}
+	return nil
+}
+
+func getPerfReplicationGroupMembersQuery(ctx context.Context, db *sql.DB, logger log.Logger) string {
```

Comment:
I'm not a fan of this method. For sure not with the `activeQuery` global variable. This is not thread safe.

We normally like to use specific columns when hitting these tables, but in this case it might make sense to use a `SELECT *` and figure out what is available based on the `Columns()` method.

Reply:
@nirgilboa just do like I did in PR, `select *` and parse the selected columns we are interesting in as string (in my PR they all were float64). You can see parsing of slave status as well for reference.

Reply:
Yes thank you both - making these changes now